appium/lib/screen-object.rb

```diff
@@ -69,82 +69,97 @@ def enter
     #pending implementation
   end
 
-  def scroll_down_find(locator,locator_value,num_loop = 15)
+  def scroll(direction)
+    size = driver.driver.manage.window.size
+    height = size.height
+    width = size.width
+    if direction != :up && direction != :down && direction != :left && direction != :right
+      CXA.output_text 'Only upwards and downwards left right scrolling is supported for now'
+    end
+    if direction == :up
+      Appium::TouchAction.new(driver).swipe(start_x: (width / 2).to_int, start_y: (height / 4).to_int, end_x: (width / 2).to_int, end_y: (height/2) - 100).perform
+    elsif direction == :down
+      Appium::TouchAction.new(driver).swipe(start_x: (width / 2).to_int, start_y: (height / 4).to_int, end_x: (width / 2).to_int, end_y: 100).perform
+    elsif direction == :left
+      Appium::TouchAction.new(driver).swipe(start_x: (width - 600).to_int, start_y: (height / 2).to_int, end_x: (width/2).to_int, end_y: height / 2).perform
+    else direction == :right
+    Appium::TouchAction.new(driver).swipe(start_x: 752, start_y: (height / 2).to_int, end_x: (width - 600).to_int, end_y: height / 2).perform
+    end
+  end
+
+  def scroll_find(locator, post_timeout = 2, direction = :down, num_loop = 15)
     scr = driver.window_size
-    screenHeightStart = (scr.height) * 0.5
-    scrollStart = screenHeightStart.to_i
-    screenHeightEnd = (scr.height) * 0.2
-    scrollEnd = screenHeightEnd.to_i
+    x = scr['width'] / 4
+    y = scr['height'] / 2.8
     for i in 0..num_loop
       begin
-        if (driver.find_element(locator,locator_value).displayed?)
+        if (driver.find_element(locator).displayed?)
```

Comment:
This if statement can easily be a one-liner since it's a guard clause I think it would be appropriate.